clean-abap/CleanABAP.md

```diff
@@ -1063,6 +1064,32 @@ LOOP AT my_table REFERENCE INTO DATA(line).
 ENDLOOP.
 ```
 
+### Avoid unnecessary table reads]
+
+> [Clean ABAP](#clean-abap) > [Content](#content) > [Tables](#tables) > [This section](#avoid-unnecessary-talbe-reads)
+
+In the case where you need to get a row identified by a key and it is highly
+unlikely that the key is missing, prefer `TRY read CATCH re-RAISE` above
+`IF line_exists( ... ) raise ENDIF. read`
+
+```ABAP
+TRY.
+    DATA(row) = my_table[ key = input ].
+  CATCH cx_sy_itab_line_not_found.
+    RAISE EXCEPTION NEW /clean/my_data_not_found( ).
+ENDTRY.
+```
+
+which is more efficient than
+
+```ABAP
+IF not line_exists( my_table[ key = intup ] ).
+  RAISE EXCEPTION NEW /clean/my_data_not_found( ).
+ENDTRY.
+
+DAT(row) = my_table[ key = intup ]
```

Comment:
Syntax error `DAT` -> `DATA` plus add `.`

plus consider upper casing "not" to have it consistently pretty printed

Reply:
Thanks, amended!

Reply:
plus "intup" -> "input"

Reply:
:D D'oh, no comment ... amended. Thanks!